cpp/daal/src/algorithms/dtrees/gbt/classification/gbt_classification_train_dense_default_impl.i

```diff
@@ -139,18 +139,15 @@ public:
             algorithmFPType * p  = bUseTLS ? lsData.local() : buf;
             const size_t iSample = (sampleInd ? sampleInd[i] : i);
             getSoftmax(f + _nClasses * iSample, p);
-            PRAGMA_FORCE_SIMD
+            PRAGMA_OMP_SIMD
             PRAGMA_VECTOR_ALWAYS
             for (size_t k = 0; k < _nClasses; ++k)
             {
                 const algorithmFPType pk = p[k];
                 const algorithmFPType h  = algorithmFPType(2.) * pk * (algorithmFPType(1.) - pk);
                 algorithmFPType * gh_ik  = gh + 2 * (k * nRows + iSample);
+                gh_ik[0]                 = (size_t(y[iSample]) == k) ? (pk - algorithmFPType(1.)) : pk;
                 gh_ik[1]                 = h;
```

Comment:
does it make sense to compute gh_ik[1] as algorithmFPType(2.) * pk * (algorithmFPType(1.) - pk); instead of additional variable h?